### ISSUE TITLE
Symfony2 and Doctrine2 entity manager issue after first test

### ISSUE THREAD

**Issue by AlexStansfield** [Codeception/Codeception#2025]
Been having an issue using the Doctrine2 and Symfony2 modules together.

It looks like even though it's persisting the entity manager between requests something is going wrong.

Basically the first test runs fine, the Entities I've created using `$I->haveInRepository` are found when we make the symfony request. However the second test (even though in the example I give below is the exact same test) fails to find them.

I know the Doctrine2 module is creating the entities inside the transaction as I `find()` one of them as part of the data setup. All I can think is there is some extra service that needs to be persisted between symfony requests that isn't being so any doctrine requests being done in the following tests are outside of the transaction that the Doctrine2 module has created.

**EDIT:** To save reading through my code examples I created a project to reproduce the issue, see the 3rd comment (https://github.com/Codeception/Codeception/issues/2025#issuecomment-109953529)

My test file is:

```
<?php
use \ApiTester;

class TokenCest
{
    public function _before(ApiTester $I)
    {
        $em = \Codeception\Module\Doctrine2::$em;
        $customerId = $I->haveInRepository(
            'Ubiquity\Bundle\ConnectBundle\Entity\Customer',
            [
                'id' => 'codeception-customer',
                'name' => 'Codeception Test',
                'added' => new \DateTime('2015-01-01'),
                'details' => 'Customer Details'
            ]);
        $customer = $em->find('UbiquityConnectBundle:Customer', $customerId);
        $I->haveInRepository(
            'Ubiquity\Bundle\ConnectBundle\Entity\Client',
            [
                'name' => 'codeception-client',
                'secret' => 'thisissecret',
                'customer' => $customer,
                'allowedGrantTypes' => ['password']
            ]);
        $I->haveInRepository(
            'Ubiquity\Bundle\ConnectBundle\Entity\User',
            [
                'username' => 'codeception-user',
                'email' => 'user@codeception.com',
                'customer' => $customer,
                'salt' => '478c1c486c1a8bc36d888f1f6de25749',
                'password' => 'I8YhzmKq77ewX4aGpt5iUyC96lCAnazvoDQpi0irkxbqfghRLw/4QRDkyEsjoykkD2/Wj4F8XoitJhdWyfXsCA=='
            ]
        );
    }

    public function _after(ApiTester $I)
    {
    }

    // tests
    public function passwordGrantTest(ApiTester $I)
    {
        $I->wantTo('get an access token with a password');
        $I->sendPOST('token', [
            'grant_type' => 'password',
            'client_id' => 'codeception-client',
            'client_secret' => 'thisissecret',
            'username' => 'codeception-user',
            'password' => 'password'
        ]);
        $I->seeResponseCodeIs(200);
        $I->seeResponseIsJson();
        $I->seeResponseContainsJson(['token_type' => 'bearer', 'expires_in' => 3600]);
    }

    // tests
    public function password2GrantTest(ApiTester $I)
    {
        $I->wantTo('get an access token with a password');
        $I->sendPOST('token', [
            'grant_type' => 'password',
            'client_id' => 'codeception-client',
            'client_secret' => 'thisissecret',
            'username' => 'codeception-user',
            'password' => 'password'
        ]);
        $I->seeResponseCodeIs(200);
        $I->seeResponseIsJson();
        $I->seeResponseContainsJson(['token_type' => 'bearer', 'expires_in' => 3600]);
    }
}
```

my api.suite.yml is:

```
class_name: ApiTester
modules:
    enabled: [Symfony2, Doctrine2, REST, ApiHelper]
    config:
        Symfony2:
            app_path: 'app'
            environment: 'test'
        Doctrine2:
            depends: Symfony2
            connection_callback: [AFakeValueHere]
            cleanup: true
        REST:
            url: http://connect.ubiquitysaas/
```

I'm using codeception 2.0.14, doctrine orm 2.4.7, doctrine dbal 2.4.4 and symfony 2.6.7

**Comment by AlexStansfield** [comment]
Got the tests to work when I changed a section of Codeception\Lib\Connector\Symfony2 from:

    protected function doRequest($request)
    {
        $services = [];
        if (self::$hasPerformedRequest) {
            $services = $this->persistServices();
            $this->kernel->shutdown();
        } else {
            self::$hasPerformedRequest = true;
        }
        $this->kernel->boot();

to 

    protected function doRequest($request)
    {
        $services = [];
        if (self::$hasPerformedRequest) {
            $services = $this->persistServices();
            //$this->kernel->shutdown();
        } else {
            self::$hasPerformedRequest = true;
        }
        $this->kernel->boot();

Obviously this is a hack and not a solution but makes me think it's definitely something not being persisted between requests and gets wiped out when the kernel is shutdown.

**Comment by AlexStansfield** [comment]
I was starting to think maybe it was something specific to my project so after spending ages debugging stuff I decided quickest thing would be to create a brand new project.

I made a new project with symfony 2.6.7 and codeception 2.0.14. Created a single entity and one controller that just takes an ID, finds the entity from the entitymanager and returns the data as JSON if it exists. Then created an ApiTester Cest that has two tests that do exactly the same test.

The first succeeds and the seconds fails.

You can check it out here: https://github.com/AlexStansfield/codeception-test

**Comment by AlexStansfield** [comment]
So interesting thing happened. Decided to add more tests (all doing the same test) and found that only the 2nd one fails.

Have updated the repo I gave above with the extra tests (https://github.com/AlexStansfield/codeception-test/blob/1c8c489f4b412def6f43347dfd3c23f2e6faeba9/tests/api/EntityCest.php). This is the result:

    Test doctrine #1 (EntityCest::tryToTest)                                       Ok
    Test doctrine #2 (EntityCest::trySecondTimeToTest)                             Fail
    Test doctrine #3 (EntityCest::tryThirdTimeToTest)                              Ok
    Test doctrine #4 (EntityCest::tryFourthTimeToTest)                             Ok
    Test doctrine #5 (EntityCest::tryFifthTimeToTest)                              Ok

**Comment by AlexStansfield** [comment]
Ok so my work around for now is to add a "fake" second test. It just does a `$I->sendPOST('');` and nothing else. Then everything after works fine.

**Comment by AlexStansfield** [comment]
I found the problem extended further than between tests but also when you are making more than 1 request in a test. We'd lose the doctrine transaction after the first transaction.

I fixed it using the change I posted in the PR. But I'm really not sure if this is the correct/best solution.

**Comment by raistlin** [comment]
This issue should have been fixed by #2968
@AlexStansfield could you check if it's still causing issues to you ?

**Comment by mmeloni** [comment]
I have same problem again. Codeception version 2.3.2. Em is closed if 2 there are 2 post in which 1 throw a doctrine exception.
